api/internal/handler/cache_verify/cache_verify.go

```diff
@@ -0,0 +1,155 @@
+	Scripts       []compareResult `json:"inconsistent_scripts"`
+	GlobalPlugins []compareResult `json:"inconsistent_global_plugins"`
+	PluginConfigs []compareResult `json:"inconsistent_plugin_configs"`
+	ServerInfos   []compareResult `json:"inconsistent_server_infos"`
+}
+
+var infixMap = map[store.HubKey]string{
+	store.HubKeyConsumer:     "consumers",
+	store.HubKeyRoute:        "routes",
+	store.HubKeyService:      "services",
+	store.HubKeySsl:          "ssl",
+	store.HubKeyUpstream:     "upstreams",
+	store.HubKeyScript:       "scripts",
+	store.HubKeyGlobalRule:   "global_rules",
+	store.HubKeyServerInfo:   "data_plane/server_info",
+	store.HubKeyPluginConfig: "plugin_configs",
+}
+
+func NewHandler() (handler.RouteRegister, error) {
+	return &Handler{}, nil
+}
+
+func (h *Handler) ApplyRoute(r *gin.Engine) {
+	r.GET("/apisix/admin/cache_verify", wgin.Wraps(h.CacheVerify))
+}
+
+func (h *Handler) CacheVerify(_ droplet.Context) (interface{}, error) {
+
+	var rs resultOuput
+	store.RangeStore(func(key store.HubKey, s *store.GenericStore) bool {
+		s.Range(context.TODO(), func(k string, obj interface{}) bool {
+			cmp, cacheValue, etcdValue := compare(k, infixMap[key], obj)
+			if !cmp {
+				if key == store.HubKeyConsumer {
+					rs.Consumers = append(rs.Consumers, compareResult{EtcdValue: etcdValue, CacheValue: cacheValue, K: k})
+				}
+				if key == store.HubKeyRoute {
+					rs.Routes = append(rs.Routes, compareResult{EtcdValue: etcdValue, CacheValue: cacheValue, K: k})
+				}
+				if key == store.HubKeyScript {
+					rs.Scripts = append(rs.Scripts, compareResult{EtcdValue: etcdValue, CacheValue: cacheValue, K: k})
+				}
+				if key == store.HubKeyService {
+					rs.Services = append(rs.Services, compareResult{EtcdValue: etcdValue, CacheValue: cacheValue, K: k})
+				}
+				if key == store.HubKeyGlobalRule {
+					rs.GlobalPlugins = append(rs.GlobalPlugins, compareResult{EtcdValue: etcdValue, CacheValue: cacheValue, K: k})
+				}
+				if key == store.HubKeyPluginConfig {
+					rs.PluginConfigs = append(rs.PluginConfigs, compareResult{EtcdValue: etcdValue, CacheValue: cacheValue, K: k})
+				}
+				if key == store.HubKeyUpstream {
+					rs.Upstreams = append(rs.Upstreams, compareResult{EtcdValue: etcdValue, CacheValue: cacheValue, K: k})
+				}
+				if key == store.HubKeySsl {
+					rs.SSLs = append(rs.SSLs, compareResult{EtcdValue: etcdValue, CacheValue: cacheValue, K: k})
+				}
+				if key == store.HubKeyServerInfo {
+					rs.ServerInfos = append(rs.ServerInfos, compareResult{EtcdValue: etcdValue, CacheValue: cacheValue, K: k})
+				}
+
+			}
+			return true
+		})
+		return true
+	})
+	return rs, nil
+}
+
+func compare(k, infix string, v interface{}) (bool, string, string) {
+	s, err := json.Marshal(v)
+	if err != nil {
+		fmt.Printf("json marsharl failed : %v\n", err)
+		return false, "", ""
+	}
+	cacheValue := string(s)
+	key := fmt.Sprintf("/apisix/%s/%s", infix, k)
+	val, err := storage.GenEtcdStorage().Get(context.TODO(), key)
```

Comment:
We could get it in batches by prefix, not every key to get once.

Reply:
> get it in batches by prefix

what does this mean? 
does it mean that we should get etcd client in the CacheVerify function,and do it once every time when we run this command ?

Reply:
See the difference between `etcdctl get  --prefix /apisix/routes` and `etcdctl get /apisix/routes/1`

Reply:
> See the difference between `etcdctl get --prefix /apisix/routes` and `etcdctl get /apisix/routes/1`

got it 
I will change it in next commit